File: hyperbolic/example/prep_grid/creat_step.m
clc;
clear all;
close all;

nz=301;
num_of_step = nz-1;
dh=-10;
for i=1:num_of_step
 step(i) = dh;
end
% for i=1:10
%   step(i) = -3;
% end
% for i=11:22
%   step(i) = step(i-1)*1.096;
% end
% for i=23:num_of_step
%   step(i) = step(22);
% end
% creat step file
file_name = '../step_file_3d.txt';
fid=fopen(file_name,'w'); % Output file name 
fprintf(fid,'# number of step\n'); 
fprintf(fid,'%d\n',num_of_step);
fprintf(fid,'# step\n'); 
for i=1:num_of_step
  fprintf(fid,'%.9e \n',step(i));
end

File: hyperbolic/example/prep_grid/creat_step.py
import sys, os
import numpy as np
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..', '..', 'plotting'))
from export_c import export_step
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt

def main():
    flag_printf = 1
    nz = 201
    num_of_step = nz - 1
    dh = -15.0
    step_vals = np.full(num_of_step, dh)

    file_name = '../step_file_3d.txt'
    export_step(step_vals, num_of_step, file_name)

    if flag_printf:
        os.makedirs('../plot', exist_ok=True)
        fig, ax = plt.subplots(figsize=(8, 4))
        ax.plot(range(1, num_of_step + 1), step_vals, 'b-', linewidth=1.5)
        ax.set_xlabel('Step index', fontweight='bold')
        ax.set_ylabel('Step value', fontweight='bold')
        ax.set_title('Step distribution', fontweight='bold')
        ax.grid(True)
        plt.tight_layout()
        plt.savefig('../plot/hyper_step.png', dpi=200, bbox_inches='tight', facecolor='white')
        plt.close(fig)
        print('Saved: ../plot/hyper_step.png')

if __name__ == "__main__":
    main()
